libtsuba/include/katana/EphemeralStoragePrefix.h

```diff
@@ -0,0 +1,67 @@
+#ifndef KATANA_LIBTSUBA_KATANA_EPHEMERALSTORAGEPREFIX_H_
+#define KATANA_LIBTSUBA_KATANA_EPHEMERALSTORAGEPREFIX_H_
+
+#include <memory>
+
+#include "katana/URI.h"
+#include "katana/file.h"
+
+namespace katana {
+
+/// EphemeralStoragePrefix uses RAII to manage ephemeral storage locations. It
+/// is constructed with a URI and when the object is destroyed it deletes all
+/// files in the directory encoded in the URI.
+///
+/// NB: it would be useful if it recursively deleted all files in all
+/// subdirectories (a 'rm -rf'-like operation).
+class EphemeralStoragePrefix {
+public:
+  ~EphemeralStoragePrefix() {
+    std::vector<std::string> files;
+    auto list_future = FileListAsync(prefix_.path(), &files);
+    if (!list_future.valid()) {
+      KATANA_LOG_WARN(
+          "unable to list files, not cleaning up ephemeral storage");
+      return;
+    }
+
+    auto list_future_res = list_future.get();
+    if (!list_future_res) {
+      KATANA_LOG_WARN(
+          "unable to list files, not cleaning up ephemeral storage: {}",
+          list_future_res.error());
+    }
+
+    std::unordered_set deletable_files(files.begin(), files.end());
+    auto delete_res = FileDelete(prefix_.path(), deletable_files);
+    if (!delete_res) {
+      KATANA_LOG_WARN(
+          "unable to delete files, not cleaning up ephemeral storage: {}",
+          delete_res.error());
+    }
+  }
```

Comment:
I just realized something you may not be aware of. On Linux/Unix, you can create, open, and immediately delete a file. The file disappears to anyone listing the directly, but it actually still exists as long as it is still open. You can use that for temp files, that you don't need to close and reopen. It avoids the need to explicitly clean up the files.

Reply:
The contract here is that `PropertyGraph` has a storage prefix that it is free to do whatever it wants with. So I don't think we will be able to use those sorts of temporary files without invasive changes to the storage layer.